Antidote/ActiveCallViewController.m

```diff
@@ -319,6 +339,84 @@ - (void)showCallPausedByFriend
     [self createFriendAvatar];
 }
 
+- (void)setVideoView:(UIView *)videoView
+{
+
+    if (_videoView == videoView) {
+        return;
+    }
+
+    if (_videoView) {
+        [_videoView removeFromSuperview];
+    }
+
+    _videoView = videoView;
+
+    if (! videoView) {
+
+        if (! self.previewLayer) {
+            [self updateCallsControlViewExpand];
+        }
+
+        return;
+    }
+
+
+    if (self.previewView) {
+        [self.view insertSubview:self.videoView belowSubview:self.previewView];
+    }
+    else {
+        [self.view addSubview:self.videoView];
+    }
+
+    [self.videoView makeConstraints:^(MASConstraintMaker *make) {
+        make.bottom.equalTo(self.endCallButton.top).with.offset(-kIndent);
+        make.top.equalTo(self.topViewContainer.bottom).with.offset(kIndent);
+        make.left.equalTo(self.view);
+        make.right.equalTo(self.view);
+    }];
+
+    [self updateCallsControlViewCondensed];
+}
+
+- (void)setPreviewLayer:(CALayer *)previewLayer
+{
+    if (_previewLayer == previewLayer) {
+        return;
+    }
+
+    if (_previewLayer) {
+        [_previewLayer removeFromSuperlayer];
+    }
+
+    _previewLayer = previewLayer;
+
+    if (! _previewLayer) {
+        [self.callControlsView setVideoButtonSelected:NO];
+        return;
```

Comment:
I think here you have to check if there is `videoView` and change type to `expand` if there isn't.

Actually I'm thinking that you can replace `updateCallsControlView{Condensed,Expand}` method with one `updateCallsControlViewType`, that will check if there is preview and viewView and change type appropriately. You can call this method in all needed places.